src/Endatix.Core/Specifications/SubmissionsByFormIdSpec.cs

```diff
@@ -12,7 +12,7 @@ namespace Endatix.Core.Specifications;
 /// TODO: [check] We can add PagedSpecification&lt;T&gt; instead of Specification&lt;T&gt; as one direction to reuse and encapsulate logic, but we need to factor in other requirements like Basic filtering/sorting/ordering + specific filtering, e.g. only ActiveForms
 /// TODO: [check] Also handle Ability to return PagedResult, which has the current page number and total count of items instead of basic list of results
 /// </summary>
-public class SubmissionsByFormIdSpec : Specification<Submission>
+public class SubmissionsByFormIdSpec : Specification<Submission, Submission>
```

Comment:
My recommendation is to change the class definition from
`public class SubmissionsByFormIdSpec : Specification<Submission, Submission>`

to
`public class SubmissionModelsByFormIdSpec : Specification<Submission, SubmissionModel>`

Idea is to add to the domain logic that the list requirements only need part of the Submission. Maybe as we move to a vertically sliced approach where all the layers will be in the same assembly that will be more natural too